packages/beacon-node/src/chain/validatorMonitor.ts

```diff
@@ -754,46 +756,56 @@ export function createValidatorMonitor(
           summary.attestationAggregateInclusions
         );
-        metrics?.validatorMonitor.prevEpochAttestationBlockInclusions.observe(summary.attestationBlockInclusions);
+        validatorMonitorMetrics?.prevEpochAttestationBlockInclusions.observe(summary.attestationBlockInclusions);
         if (summary.attestationMinBlockInclusionDistance !== null) {
-          metrics?.validatorMonitor.prevEpochAttestationBlockMinInclusionDistance.observe(
+          validatorMonitorMetrics?.prevEpochAttestationBlockMinInclusionDistance.observe(
             summary.attestationMinBlockInclusionDistance
           );
         }
 
         // Blocks
-        metrics?.validatorMonitor.prevEpochBeaconBlocks.observe(summary.blocks);
+        validatorMonitorMetrics?.prevEpochBeaconBlocks.observe(summary.blocks);
         if (summary.blockMinDelay !== null)
-          metrics?.validatorMonitor.prevEpochBeaconBlocksMinDelaySeconds.observe(summary.blockMinDelay);
+          validatorMonitorMetrics?.prevEpochBeaconBlocksMinDelaySeconds.observe(summary.blockMinDelay);
 
         // Aggregates
-        metrics?.validatorMonitor.prevEpochAggregatesTotal.observe(summary.aggregates);
+        validatorMonitorMetrics?.prevEpochAggregatesTotal.observe(summary.aggregates);
         if (summary.aggregateMinDelay !== null)
-          metrics?.validatorMonitor.prevEpochAggregatesMinDelaySeconds.observe(summary.aggregateMinDelay);
+          validatorMonitorMetrics?.prevEpochAggregatesMinDelaySeconds.observe(summary.aggregateMinDelay);
 
         // Sync committee
         prevEpochSyncCommitteeHits += summary.syncCommitteeHits;
         prevEpochSyncCommitteeMisses += summary.syncCommitteeMisses;
 
         // Only observe if included in sync committee to prevent distorting metrics
         if (validatorInSyncCommittee) {
-          metrics?.validatorMonitor.prevEpochSyncSignatureAggregateInclusions.observe(
+          validatorMonitorMetrics?.prevEpochSyncSignatureAggregateInclusions.observe(
             summary.syncSignatureAggregateInclusions
           );
         }
       }
 
-      metrics?.validatorMonitor.validatorsInSyncCommittee.set(validatorsInSyncCommittee);
-      metrics?.validatorMonitor.prevEpochSyncCommitteeHits.set(prevEpochSyncCommitteeHits);
-      metrics?.validatorMonitor.prevEpochSyncCommitteeMisses.set(prevEpochSyncCommitteeMisses);
+      validatorMonitorMetrics?.validatorsInSyncCommittee.set(validatorsInSyncCommittee);
+      validatorMonitorMetrics?.prevEpochSyncCommitteeHits.set(prevEpochSyncCommitteeHits);
+      validatorMonitorMetrics?.prevEpochSyncCommitteeMisses.set(prevEpochSyncCommitteeMisses);
     },
   };
+
+  if (metricsRegister) {
+    // Register a single collect() function to run all validatorMonitor metrics
+    validatorMonitorMetrics?.validatorsConnected.addCollect(() => {
+      const clockSlot = getCurrentSlot(config, genesisTime);
+      validatorMonitor.scrapeMetrics(clockSlot);
+    });
+  }
```

Comment:
don't need to check for metrics registry here, this should do
```suggestion
  // Register a single collect() function to run all validatorMonitor metrics
  validatorMonitorMetrics?.validatorsConnected.addCollect(() => {
    const clockSlot = getCurrentSlot(config, genesisTime);
    validatorMonitor.scrapeMetrics(clockSlot);
  });
```